### PR TITLE
[v2] Import reference entity records as product option values

### PR DESCRIPTION
**Pull request by TheGrimmChester**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?   | no

### CODE REVIEW

**Comment by TheGrimmChester** [comment]
TODO:

- [ ] code cleanup
- [ ] tests

File: src/Processor/ProductOptionValue/ReferenceEntityOptionValuesProcessor.php [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    {
+        return 90;
+    }
+
+    public function support(AttributeInterface $attribute, ProductOptionInterface $productOption, array $context = []): bool
+    {
+        return ReferenceEntityAttributeType::TYPE === $attribute->getType() && $this->isEnterprise();
+    }
+
+    private function isEnterprise(): bool
```
Comment (oallain) [inline src/Processor/ProductOptionValue/ReferenceEntityOptionValuesProcessor.php:79]: IMO, this function shoundn't be here.

File: src/Processor/ProductOptionValue/ReferenceEntityOptionValuesProcessor.php [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                $productOptionValueTranslation->setLocale($locale);
+
+                $this->entityManager->persist($productOptionValueTranslation);
+            }
+
+            $productOptionValueTranslation->setValue($translation['data']);
+        }
+    }
+
+    private function getLocales(): iterable
```
Comment (oallain) [inline src/Processor/ProductOptionValue/ReferenceEntityOptionValuesProcessor.php:166]: IMO, this function shoundn't be here.

File: src/Task/Attribute/CreateUpdateEntityTask.php [hunk 3]
```diff
@@ -105,50 +116,60 @@ public function __invoke(PipelinePayloadInterface $payload): PipelinePayloadInte
...
         }
 
-        if (!$payload->getResources() instanceof ResourceCursorInterface) {
-            throw new NoAttributeResourcesException('No resource found.');
-        }
-
         try {
             $excludesAttributes = $this->excludedAttributesProvider->getExcludedAttributes();
-
+            $isEnterprise = $apiConfiguration->isEnterprise() ?? false;
```
Comment (oallain) [inline src/Task/Attribute/CreateUpdateEntityTask.php:121]: Maybe you should use https://github.com/synolia/SyliusAkeneoPlugin/pull/73/files#diff-1cd56ef8a27450489cf1349bb626d3b3a1c4c8a32180450581cd7916a3d54754R79

File: src/Transformer/AttributeOptionValueDataTransformer.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        return \strtolower(\sprintf(
+            '%s%s',
+            self::AKENEO_PREFIX,
+            $value
+        ));
+    }
+
+    public function reverseTransform(string $value): string
+    {
+        return \str_replace(\sprintf('%s', self::AKENEO_PREFIX), '', $value);
```
Comment (oallain) [inline src/Transformer/AttributeOptionValueDataTransformer.php:22]:
```suggestion
        return \str_replace(self::AKENEO_PREFIX, '', $value);
```

File: tests/PHPUnit/Task/Attribute/CreateUpdateDeleteTaskTest.php [hunk 5]
```diff
@@ -19,64 +18,48 @@ final class CreateUpdateDeleteTaskTest extends AbstractTaskTest
...
 
         /** @var \Sylius\Component\Product\Model\ProductAttribute $colorProductAttribute */
-        $colorProductAttribute = self::$container->get('sylius.repository.product_attribute')->findOneBy(['code' => 'color']);
+        $colorProductAttribute = $this->getContainer()->get('sylius.repository.product_attribute')->findOneBy(['code' => 'color']);
         $this->assertNotNull($colorProductAttribute);
         $this->assertEquals('Couleur', $colorProductAttribute->getTranslation('fr_FR')->getName());
         $this->assertEquals('Color', $colorProductAttribute->getTranslation('en_US')->getName());
     }
-
-    public function testDeleteTask(): void
```
Comment (oallain) [inline tests/PHPUnit/Task/Attribute/CreateUpdateDeleteTaskTest.php:58]: We assume this deletion?

**Review by oallain — COMMENTED** [review]
Thanks for the great work, I just made few small feedbacks.

File: tests/PHPUnit/Task/Attribute/CreateUpdateDeleteTaskTest.php (hunk 5 above)
Comment (TheGrimmChester) [reply tests/PHPUnit/Task/Attribute/CreateUpdateDeleteTaskTest.php:58]: yes, it was present but never used. We only do create/update for attributes.

File: src/Processor/ProductOptionValue/ReferenceEntityOptionValuesProcessor.php (hunk 1 above)
Comment (TheGrimmChester) [reply src/Processor/ProductOptionValue/ReferenceEntityOptionValuesProcessor.php]: fixed in commit 3f521e9319e96e2aca784b1d4b43965bccca3fb5

File: src/Task/Attribute/CreateUpdateEntityTask.php (hunk 3 above)
Comment (TheGrimmChester) [reply src/Task/Attribute/CreateUpdateEntityTask.php]: fixed in commit 3f521e9319e96e2aca784b1d4b43965bccca3fb5

File: src/Processor/ProductOptionValue/ReferenceEntityOptionValuesProcessor.php (hunk 2 above)
Comment (TheGrimmChester) [reply src/Processor/ProductOptionValue/ReferenceEntityOptionValuesProcessor.php]: fixed in commit bf87467577e17a7eb1ed6df554bb360c3514c478

**Review by oallain — APPROVED** [review]
Nice work !

**Pull request merged**